### DISCUSSION TITLE
YELLOW icon instead of GREEN when the last ping was SUCCESSFUL

### DISCUSSION THREAD

**Discussion by pmatuszy** [healthchecks/healthchecks#789] [General]
Hi

I noticed a couple of times that I have yellow icons instead of green even when the last ping was successful:
here is the example:
![image](https://user-images.githubusercontent.com/58336955/216354659-f76c16c5-4c22-4e27-b178-e21d7ee527ad.png)

here is the ping status:
![image](https://user-images.githubusercontent.com/58336955/216457133-5b77b774-77bd-45ad-b80f-95312a81491f.png)

Have you seen it in any other installations?
Regards
Paul

**Comment by cuu508** [comment]
When you configure check's schedule using a cron expression, Healthchecks expects to receive a ping **at or after the datetime matching the expression**. If the ping arrives earlier, it does not count. What I think happens here, is:

* the schedule is "at 7:01, at 8:01, at 9:01, ...., at 22:01"
* a ping arrived at 15:00 – this is a little too early to count as a ping for 15:01
* healthchecks still expects to receive a ping at or after 15:01, so from 15:01 until the next ping it shows the status as yellow/orange 

One way to fix this is to adjust the cron expression a little bit – if you know the pings could arrive early at xx:00 instead of xx:01, set the schedule to "0 7-22 * * *".

Another thing to check is if the clock on the client is accurate (e.g. if a ntp service is working).
